HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc

```diff
@@ -232,12 +232,15 @@ std::shared_ptr<HLTriggerJSONMonitoringData::run> HLTriggerJSONMonitoring::globa
     for (auto i = 0u; i < triggersSize; ++i) {
       rundata->posL1s[i] = -1;
       rundata->posPre[i] = -1;
-      auto const& moduleLabels = rundata->hltConfig.moduleLabels(i);
+      auto const trigNameIndx = rundata->indicesOfTriggerPaths[i];
+      auto const& moduleLabels = rundata->hltConfig.moduleLabels(trigNameIndx);
       for (auto j = 0u; j < moduleLabels.size(); ++j) {
-        auto const& label = rundata->hltConfig.moduleType(moduleLabels[j]);
-        if (label == "HLTL1TSeed")
+        auto const& moduleLabel = moduleLabels[j];
+        auto const& moduleType = rundata->hltConfig.moduleType(moduleLabel);
+        //the logic here is that it finds the first non ignored HLTL1TSeed module, ignored modules have - at the start of their name
+        if (rundata->posL1s[i] == -1 && moduleLabel.rfind('-', 0) != 0 && moduleType == "HLTL1TSeed") {
```

Comment:
its a good point. Its definitely something to consider. I think actually we could adjust the logic to the following

last non ignored HLTL1Seed or TriggerResultsFilter (possibly checking if its going for L1 seeds) before the prescale module.  

Now thinking about the reason for going last before prescale module means that you could effectively AND two modules this way and I think that fits the spirit of the info we are trying to convey.  Not sure why you would do this but you could and this would account for that case while still working in the normal one

Reply:
Yes, agreed. Regarding the "before the prescale module" condition, I think we could still get the "right" info without this condition in all realistic cases (I suspect this condition was avoided intentionally in the current implementation). I don't know what "the info we are trying to convey" should be in edge cases, e.g. [*].

> TriggerResultsFilter (possibly checking if its going for L1 seeds)

This should be possible (using, for example, `HLTConfigProvider::modulePSet`), but the exact condition is maybe a bit tricky to craft (e.g. `usePathStatus=False`, plus `l1tResults != ''`, plus `triggerConditions[i] 'contains' "L1_"`, or similar).

Being pragmatic, I would be okay with this PR as it is (including it asap in `12_4_X` b/c of the bugfix), leaving the above improvements to a future PR. If you prefer to make these updates here, that's also okay, of course.

[*] A (non-realistic) Path with "HLTL1TSeed(1) + HLTL1TSeed(2) + HLTPrescaler + [..] + HLTL1TSeed(3) + [..]": would we take (2) or (3)?

Reply:
So I would say we take a little time to get this correct, the bug is not critical and is not exposed as long as people keep their path before dataset paths which we normally do.  But if you prefer to push ahead quickly , I'm fine with that. 

To me a correctly designed path has L1 selection -> pass prescale -> additional selection  (which may be or may not be L1 based but to me now counts as selection logic rather than L1 selection). I admit others may think differently. 

And also I think the main use case of these numbers is seeing if the HLTPrescaler rejected anything.  So in your example I would say  HLTL1Seed(1) + HLTL1TSeed(2) are actually a single L1 selection ANDed together.  HLTL1Seed(3) is trigger selection logic and is no longer a "pure" L1 selection.  So I would take filter 2. 

If we were to take (3), I'm not sure how meaningful this number would be as its events passing L1  selection + other selection. 

We could also redefine the numbers to be 

events passing before prescaler 
events passing after prescaler
events passing path

so we take any filter before the HLT prescaler.  What do you think?

Reply:
> To me a correctly designed path has L1 selection -> path prescale -> additional selection (which may be or may not be L1 based but to me now counts as selection logic rather than L1 selection).

@Sam-Harper FWIW I agree with the logic you describe.

Reply:
thanks Andrea, so would you be fine with  the logic

events passing before prescaler
events passing after prescaler
events passing path

which that reduces to. Or do you think we should explicitly look for L1 seed modules before the prescaler?

Reply:
Thanks for explaining; this makes perfect sense to me. "events passing before prescaler" would also allow us to keep the implementation very simple (no need to check for "HLTL1TSeed", nor "TriggerResultsFilter" and its PSet, not even the 'ignore' character; the only 'hard-coded' name left in the plugin would be "HLTPrescaler").

Reply:
I notify @fwyzard of this discussion, since he knows the purpose of this plugin better than me.

Reply:
it would also detect trigger type then (due to the trigger type filter) so for example HLT_Random is currently 

| HLT_Random_v3 | 848303444 | 90761 | 90761 |  

but thatis the full L1 input rate while it would make sense for this to be just the randoms triggers which this fix would make.

Reply:
> events passing before prescaler
> events passing after prescaler
> events passing path

Yes, I agree this simplification of the logic would both make the code simpler and the logic more robust, for example it would include the "trigger type" in the "L1 selection" (even though technically it's part of the TCDS selection, the difference doesn't really matter for the HLT).